File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& nums, int k) {
        unordered_map<int,int> m;
        int res = 0,cnt = 0,s = 0;
        for(int i = 0;i<nums.size();i++){
            while(m[nums[i]] == k){
                res = max(res,cnt);
                cnt--;
                m[nums[s++]]--;
            }
            m[nums[i]]++;
            cnt++;
        }
        res = max(res,cnt);
        return res;
    }
};

File: 2958-length-of-longest-subarray-with-at-most-k-frequency/2958-length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        dic = defaultdict(int)
        res,j,cnt = 0,0,0
        for i,x in enumerate(nums):
            dic[x] += 1
            while dic[x] > k:
                res = max(cnt,res)
                dic[nums[j]] -= 1
                j += 1
            cnt = i - j + 1
        return max(res,cnt)
